### DISCUSSION TITLE
关于DNS设置的疑问

### DISCUSSION THREAD

**Discussion by junjun1121** [immortalwrt/homeproxy#68] [Q&A]
先说Sorry因为我觉得应该是开在Discuss里的但天灵大佬没有开…
主要是有些许疑惑，刚开始尝试使用
请教一下，如果路由开启mosdns的情况下，
DNS 服务器/国内 DNS 服务器 应该选择【禁用】还是选择【使用 WAN 下发的 DNS】才会把解析权交给mosdns使用呢？
还是说……都不会？

**Comment by muink** [comment]
都不会, 需要在`DNS 服务器`指定mosdns的监听地址和端口

**Comment by jh0072110** [comment]
> 都不会, 需要在`DNS 服务器`指定mosdns的监听地址和端口

端口该如何指定，输入:或者#都会被标红唉

**Comment by muink** [comment]
好像是的, 只允許輸入ip 😿

**Comment by jh0072110** [comment]
> > > 都不会, 需要在`DNS 服务器`指定mosdns的监听地址和端口
> > 
> > 
> > 端口该如何指定，输入:或者#都会被标红唉
> 
> 直接mosdns dns转发打开 HomeProxy 国内外dns指向本地127.0.0.1

这不太合理吧，不指端口是不是有点愣来了…

**Comment by 1715173329** [comment]
全局模式、自定义模式，HP 会接管所有 DNS 请求。
GFWList 模式，HP 只接管 GFWList 内域名 DNS 请求。
绕过大陆模式，HP 只接管 GFWList 内域名 DNS 请求；开启国内 DNS 后，接管所有 DNS 请求。
代理大陆模式，HP 只接管 ChinaList 内域名 DNS 请求。

sing-box 的文档未声明支持非标准端口，所以 HP 不会支持。
HP 也不会考虑支持 MosDNS 等其他 DNS 服务，折腾 DNS 没有任何意义。

**Reply by lonelysoul** [reply]
使用绕过大陆模式，hp只接管gfwlist域名的dns请求，这部分设置为Google等；国内禁用。
同时，安装mosdns，作为HP不解析的域名的兜底解析服务，启用“防止dns泄漏”，会把fallback的全都转移到远程解析。
以上两部完成后，预测：gfwlist之外的未知的国内外域名全部走远程dns（如谷歌等），只有现有规则内确定的站（HP规则&MOSDNS规则）走国内dns（国内dns由mosdns指定了）。
这样配置后，在访问Netflix等检测dns网站时，大概率就不会检测到dns泄漏的代理使用了。
以上为粗浅理解。

**Comment by ZhongMoDao** [comment]
> 先说Sorry因为我觉得应该是开在Discuss里的但天灵大佬没有开… 主要是有些许疑惑，刚开始尝试使用 请教一下，如果路由开启mosdns的情况下， DNS 服务器/国内 DNS 服务器 应该选择【禁用】还是选择【使用 WAN 下发的 DNS】才会把解析权交给mosdns使用呢？ 还是说……都不会？

我试了是没问题的，要开启mosdns的dns劫持，然后正常设置其他，再把home里面的国内外DNS全都设置成使用WAN下发的DNS就可以了，我单纯因为有去广告需求，其他那些插件死的死，废的废，只能用mosdns了。
